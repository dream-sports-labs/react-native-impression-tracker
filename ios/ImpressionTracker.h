
#import <ImpressionTracker/RNImpressionTrackerSpec.h>

@interface ImpressionTracker : NSObject <NativeImpressionTrackerSpec>

@end
