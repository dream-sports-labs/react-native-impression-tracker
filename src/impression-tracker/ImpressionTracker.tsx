import { useCallback, FC } from 'react';

import { View } from 'react-native';

import { ImpressionTrackerContext } from './hooks/useImpressionTracker';
import { useVisibilityTracker } from './hooks/useVisibilityTracker';
import { AdsClickedInterface, ImpressionTrackerProps } from './interface';

const ImpressionTracker: FC<ImpressionTrackerProps> = ({
  children,
  rootMargin,
  desiredImpressionViewability = 0.5,
  desiredImpressionTime = 1000,
  onImpressionTrigger,
  onRealEstateClicked,
}) => {
  const visibilityTrackerView = useVisibilityTracker({
    rootMargin,
    desiredImpressionViewability,
    desiredImpressionTime,
    onImpressionTrigger,
  });

  const handleClickForImpression = useCallback(
    (customParams: AdsClickedInterface) => {
      onRealEstateClicked?.(customParams);
    },
    [onRealEstateClicked]
  );

  return (
    <ImpressionTrackerContext.Provider value={{ handleClickForImpression }}>
      <View
        collapsable={false}
        ref={visibilityTrackerView}
        testID="impression-root-view"
      >
        {children}
      </View>
    </ImpressionTrackerContext.Provider>
  );
};

export { ImpressionTracker };
