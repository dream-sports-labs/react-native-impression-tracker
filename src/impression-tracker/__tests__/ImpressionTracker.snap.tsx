import { render } from '@testing-library/react-native';

import { Text } from 'react-native';
import { ImpressionTracker } from '../ImpressionTracker';

describe('ImpressionTracker Snapshot', () => {
  it('should match the snapshot', () => {
    const { toJSON } = render(
      <ImpressionTracker onImpressionTrigger={jest.fn()}>
        <Text>Snapshot Test</Text>
      </ImpressionTracker>
    );

    // Compare the rendered output to a previously saved snapshot
    expect(toJSON()).toMatchSnapshot();
  });
});
