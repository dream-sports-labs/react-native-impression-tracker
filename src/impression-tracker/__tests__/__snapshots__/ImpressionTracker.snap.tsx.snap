// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImpressionTracker Snapshot should match the snapshot 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
        "position": "relative",
      }
    }
  >
    <View
      onTouchStart={[Function]}
      style={
        {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={false}
        testID="impression-root-view"
      >
        <Text
          allowFontScaling={false}
          style={
            [
              {
                "fontFamily": "Roboto-Medium",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0,
                "lineHeight": 20,
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              undefined,
              {
                "color": "#202C3D",
              },
            ]
          }
        >
          Snapshot Test
        </Text>
      </View>
    </View>
  </View>
</View>
`;
